packages/block-editor/src/components/index.js

```diff
@@ -15,6 +15,7 @@ export { default as BlockFormatControls } from './block-format-controls';
 export { default as BlockIcon } from './block-icon';
 export { default as BlockNavigationDropdown } from './block-navigation/dropdown';
 export { default as __experimentalBlockNavigationList } from './block-navigation/list';
+export { default as __experimentalBlockPatternPicker } from './block-pattern-picker';
```

Comment:
I guess we don't need this to be exported and experimental since it's something built-in for block-editor package.

Reply:
It depends on how it ends up. In https://github.com/WordPress/gutenberg/pull/18343 I use this component from the `@wordpress/block-library` package. In the long run, I hope we can make it implementation detail of `BlockEdit` when block patterns are registered, but there might be some challenges in making it work this way like how to detect whether a block is in the placeholder mode. I also need to validate integration with the Table block this week. Let's revisit afterward. 

Wrapping up, I 100 % agree with you, but I find it easier to keep it exposed for the time being for my experiments. I can move it to #18343 if that will help to merge this PR sooner.

Reply:
> how to detect whether a block is in the placeholder mode

I agree that this is the challenge, I remember discussing this with @mcsf and @mtias at some poinit (forgot the outcome of that discussion). but I really think we should build this as an implementation detail of BlockEdit (or other block editor components) from the start. Registering patterns for blocks, should be enough to tweak the behavior of the block (like styles, transforms)

Reply:
I like your vision, this is how I see it myself as well. I think @mtias might be in opposition or I didn't understand him properly when we discussed on Slack last time :)